src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NativeInteger.cs

```diff
@@ -1022,6 +1022,52 @@ void assert(string expectedType, string fieldName)
             }
         }
 
+        [Fact]
+        public void FunctionPointersWithNativeIntegerTypes()
+        {
+            var comp = CompileAndVerify(@"
+unsafe class C
+{
+    public delegate*<nint, nint, nint> F1;
+    public delegate*<System.IntPtr, System.IntPtr, nint> F2;
+    public delegate*<nint, System.IntPtr, System.IntPtr> F3;
+    public delegate*<System.IntPtr, nint, System.IntPtr> F4;
+    public delegate*<delegate*<System.IntPtr, System.IntPtr, System.IntPtr>, nint> F5;
```

Comment:
>F5 [](start = 83, length = 2)

Consider adding a case where the `nint` is in the nested function pointer, rather than in the outer one